File: 62750938-main/readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int main(void)
{
    string input = get_string("Text: ");
    int letters = 0;
    int words = 1;
    int sentences = 0;

    for (int i = 0, n = strlen(input); i < n; i++)
    {
        if (isalpha(input[i])) // Count alphabetic characters as letters
        {
            letters++;
        }
        else if (input[i] == ' ') // Count spaces as word separators
        {
            words++;
        }
        else if (input[i] == '.' || input[i] == '!' || input[i] == '?') // Sentence terminators
        {
            sentences++;
        }
    }

    float l = ((float) letters / words) * 100;
    float s = ((float) sentences / words) * 100;

    float index = (0.0588 * l) - (0.296 * s) - 15.8;

    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", (int) round(index));
    }
}

File: 62750938-main/sentimental-readability/readability.py
import cs50

input = cs50.get_string("Text: ")

letters = 0
words = 1  # to add the first word, that doesnt have a space before
sentences = 0


for char in input:
    if char.isalpha():
        letters += 1

    if char == ' ':
        words += 1

    if char == '.' or char == '!' or char == '?':
        sentences += 1

l = (letters / words) * 100
s = (sentences / words) * 100

cli = (0.0588 * l) - (0.296 * s) - 15.8

if cli < 1:
    print("Before Grade 1")
elif cli > 16:
    print("Grade 16+")
else:
    print(f"Grade {round(cli)}")
